built_value_generator/lib/src/plugin/plugin.dart

```diff
@@ -0,0 +1,108 @@
+import 'dart:async';
+
+import 'package:analyzer/context/context_root.dart';
+import 'package:analyzer/file_system/file_system.dart';
+// ignore: implementation_imports
+import 'package:analyzer/src/context/builder.dart';
+// ignore: implementation_imports
+import 'package:analyzer/src/dart/analysis/driver.dart';
+import 'package:analyzer_plugin/plugin/plugin.dart';
+import 'package:analyzer_plugin/protocol/protocol_common.dart' as plugin;
+import 'package:analyzer_plugin/protocol/protocol_generated.dart' as plugin;
+import 'package:built_value_generator/src/plugin/checker.dart';
+
+/// Analyzer plugin for built_value.
+///
+/// Surfaces the same errors as the generator at compile time, with fixes
+/// where possible.
+class BuiltValueAnalyzerPlugin extends ServerPlugin {
+  final Checker checker = new Checker();
+
+  BuiltValueAnalyzerPlugin(ResourceProvider provider) : super(provider);
+
+  @override
+  AnalysisDriverGeneric createAnalysisDriver(plugin.ContextRoot contextRoot) {
+    final root = new ContextRoot(contextRoot.root, contextRoot.exclude)
+      ..optionsFilePath = contextRoot.optionsFile;
+    final contextBuilder =
+        new ContextBuilder(resourceProvider, sdkManager, null)
+          ..analysisDriverScheduler = analysisDriverScheduler
+          ..byteStore = byteStore
+          ..performanceLog = performanceLog
+          ..fileContentOverlay = fileContentOverlay;
+    final result = contextBuilder.buildDriver(root);
+    result.results.listen(processResult);
+    return result;
+  }
+
+  @override
+  List<String> get fileGlobsToAnalyze => const ['*.dart'];
+
+  @override
+  String get name => 'Built Value';
+
+  // This is the protocol version, not the plugin version. It must match the
+  // version of the `analyzer_plugin` package.
+  @override
+  String get version => '1.0.0-alpha.0';
+
+  @override
+  String get contactInfo => 'https://github.com/google/built_value.dart/issues';
+
+  void processResult(AnalysisResult analysisResult) {
+    try {
+      if (analysisResult.unit == null ||
+          analysisResult.libraryElement == null) {
+        channel.sendNotification(
+            new plugin.AnalysisErrorsParams(analysisResult.path, [])
+                .toNotification());
+        return;
+      }
+      final checkResult = checker.check(analysisResult.libraryElement);
```

Comment:
Add a comment why this is a separate case then the previous one, it is not obvious to me what additional info you get by sending extra data.

Reply:
Done.